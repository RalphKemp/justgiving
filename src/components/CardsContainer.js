import React, { Component } from 'react';
import styled from 'styled-components';
import * as actions from '../actions';
import { connect } from 'react-redux';
import InfoCard from './InfoCard';
import DonationsCard from './DonationsCard';
import { media } from '../helpers/sizing';
import MyCodeLoader from '../helpers/contentLoader';
import { Transition } from 'react-spring';

const MainContainer = styled.div`
  width: 90%;
  div {
    display: flex;
    flex-direction: column;
    ${media.tablet`
      flex-direction: row;
    `};
  }
`;

class CardsContainer extends Component {
  constructor(props) {
    super(props);
    this.state = { loaded: false };
  }

  async componentDidMount() {
    await this.props.fetchCharity();
    await this.props.fetchCharityDonations();
    await this.setState({ loaded: true });
  }

  render() {
    const { charity, donations } = this.props;
    console.log(donations);
    return (
      <MainContainer>
        {this.state.loaded ? (
          <Transition
            from={{ opacity: 0 }}
            enter={{ opacity: 1 }}
            leave={{ opacity: 0 }}
          >
            {this.state.loaded
              ? styles => (
                  <div style={styles}>
                    <InfoCard charityInfo={charity} />
                    <DonationsCard donations={donations} />
                  </div>
                )
              : null}
          </Transition>
        ) : (
          <MyCodeLoader />
        )}
      </MainContainer>
    );
  }
}

function mapStateToProps({ charity, donations }) {
  return { charity, donations };
}

export default connect(
  mapStateToProps,
  actions
)(CardsContainer);
