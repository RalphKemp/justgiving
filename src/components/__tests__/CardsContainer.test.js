import React from 'react';
import { shallow } from 'enzyme';
// using curly braces to import component without state.
import { CardsContainer } from '../CardsContainer.js';

describe('CardsContainer', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<CardsContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
