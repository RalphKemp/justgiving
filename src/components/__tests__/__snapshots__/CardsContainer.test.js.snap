// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsContainer renders correctly 1`] = `
<styled.div>
  <MyCodeLoader />
</styled.div>
`;
